assets/js/query_params.js

```diff
@@ -0,0 +1,32 @@
+import DOMPurify from 'dompurify';
+
+const ALLOWED_KEYS = Object.freeze(['partner', 'partner_div']);
+
+export function storeUrlQueryParams(allowedKeys = ALLOWED_KEYS) {
+  if (typeof URLSearchParams === 'undefined' || typeof localStorage === 'undefined') {
+    return;
+  }
+
+  const urlParams = new URLSearchParams(window.location.search);
+
+  urlParams.forEach((value, key) => {
+    if (allowedKeys.includes(key)) {
+      const sanitizedValue = DOMPurify.sanitize(value);
+      localStorage.setItem(key, sanitizedValue);
```

Comment:
`localStorage` will persist forever, which could create interesting quandaries for shared computers. Maybe [`sessionStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) would be more appropriate?

Reply:
Also, besides shared computers, if a user visits the brochure site directly a few months after being redirected from the IdP, would we still want to associate the same partner agency?

Reply:
Would `sessionStorage` possibly lose the data we care about if users open articles or the contact page in new tabs?

Maybe we could set a manual "last set at" timestamp and intentionally ignore values if too old? (like manual expiration?)

Reply:
> Would `sessionStorage` possibly lose the data we care about if users open articles or the contact page in new tabs?
>
>Maybe we could set a manual "last set at" timestamp and intentionally ignore values if too old? (like manual expiration?)

Yeah, that's true. Maybe we'd be okay with that, though I could imagine some users might create new tabs as they're navigating help articles. I don't think comparing dates really gets at the original issue mentioned with shared computers.

Maybe another option is to assign a [session cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#removal_defining_the_lifetime_of_a_cookie) in JavaScript? I've never done this so I don't know if it's technically possible, but session cookies are _slightly_ more sticky than `sessionStorage` while still typically going away after the browser is closed.

Reply:
There's trade offs to both `localStorage` and `sessionStorage` as you two have pointed out. I'd like to keep it as `localStorage`. However, we can change it to `sessionStorage` if preferred. I think either `localStorage` or `sessionStorage` make sense.